packages/graphback/src/database/migrations/DataResourcesManager.ts

```diff
@@ -0,0 +1,379 @@
+// tslint:disable: await-promise
+import { Change, ChangeType } from '@graphql-inspector/core';
+import { unlinkSync } from 'fs';
+import { GlobSync } from 'glob';
+import * as Knex from 'knex';
+import { InputModelFieldContext, InputModelTypeContext } from '../../input/ContextTypes';
+import { logger } from '../../utils/logger';
+import { DatabaseContextProvider } from './DatabaseContextProvider';
+/**
+ * Represents update for data type
+ */
+export interface IDataLayerUpdate {
+  oldType: InputModelTypeContext,
+  newType: InputModelTypeContext
+}
+
+/**
+ * Interface for creating atabase resources .
+ * For example for relational databases implementors will execute DDL queries into database
+ */
+export interface IDataLayerResourcesManager {
+  /**
+   * Create resources for provided schema
+   *
+   * @param types that should be used to gather resources
+   */
+  createDatabaseResources(context: DatabaseContextProvider, types: InputModelTypeContext[]): Promise<void>;
+
+  /**
+   *  Create relations among tables in database.
+   */
+  createDatabaseRelations(context: DatabaseContextProvider, types: InputModelTypeContext[]): Promise<void>;
+
+  /**
+   * Update database resources after they are created
+   *
+   * @param updates to types
+   */
+  updateDatabaseResources(context: DatabaseContextProvider,
+    types: InputModelTypeContext[], changes: Change[]): Promise<void>;
+}
+
+/**
+ * Types of database Relations
+ * 1-1, 1:m, m:n
+ */
+enum Relation {
+  oneToOne = 'OneToOne',
+  oneToMany = 'OneToMany',
+  manyToMany = 'ManyToMany',
+}
+
+const createDBConnectionKnex = (
```

Comment:
Why this is here?

Reply:
This file always existed like this, it just shows as a new file because I moved it. See https://github.com/aerogear/graphback/blob/master/packages/graphback/src/migrations/DataResourcesManager.ts#L51